### PR TITLE
New Resource aws_elasticache_serverless_cache

### LINKED ISSUE #34590: [Enhancement]: Add support for ElastiCache Serverless
### Description

As announced by AWS, there is now a serverless option of ElastiCache. This issue is to request adding support for it.

### Affected Resource(s) and/or Data Source(s)

* aws_elasticache_replication_group
* elasticache_cluster

### Potential Terraform Configuration

_No response_

### References

- [Announcement](https://aws.amazon.com/blogs/aws/amazon-elasticache-serverless-for-redis-and-memcached-now-generally-available/)
- [API Reference](https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateServerlessCache.html)

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by meetreks**
### Description

This PR is to add a new resource aws_eelasticache_serverless to the ElasicCache Family.

### Relations

Closes #34590 

### References

[Announcement](https://aws.amazon.com/blogs/aws/amazon-elasticache-serverless-for-redis-and-memcached-now-generally-available/)
[API Reference](https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateServerlessCache.html)

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccXXX PKG=ec2

...
```
$ make testacc PKG=elasticache TESTS=TestAccServerlessElastiCache_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/elasticache/... -v -count 1 -parallel 20 -run='TestAccServerlessElastiCache_'  -timeout 360m
=== RUN   TestAccServerlessElastiCache_basic
=== PAUSE TestAccServerlessElastiCache_basic
=== RUN   TestAccServerlessElastiCache_basicRedis
=== PAUSE TestAccServerlessElastiCache_basicRedis
=== RUN   TestAccServerlessElastiCache_full
=== PAUSE TestAccServerlessElastiCache_full
=== RUN   TestAccServerlessElastiCache_fullRedis
=== PAUSE TestAccServerlessElastiCache_fullRedis
=== RUN   TestAccServerlessElastiCache_update
=== PAUSE TestAccServerlessElastiCache_update
=== RUN   TestAccServerlessElastiCache_dissapears
=== PAUSE TestAccServerlessElastiCache_dissapears
=== CONT  TestAccServerlessElastiCache_basic
=== CONT  TestAccServerlessElastiCache_fullRedis
=== CONT  TestAccServerlessElastiCache_dissapears
=== CONT  TestAccServerlessElastiCache_full
=== CONT  TestAccServerlessElastiCache_update
=== CONT  TestAccServerlessElastiCache_basicRedis
--- PASS: TestAccServerlessElastiCache_dissapears (351.69s)
--- PASS: TestAccServerlessElastiCache_basic (390.85s)
--- PASS: TestAccServerlessElastiCache_basicRedis (439.94s)
--- PASS: TestAccServerlessElastiCache_full (468.11s)
--- PASS: TestAccServerlessElastiCache_fullRedis (469.44s)
--- PASS: TestAccServerlessElastiCache_update (516.67s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/elasticache        516.814s

### CODE REVIEW

**Comment by junoriosity** [comment]
@meetreks I really would love to make use of the `aws_elasticache_serverless` resource. Many tanks for all your great support. 🙂

Could you tell me, when this will be possible?

**Comment by meetreks** [comment]
@ewbankkit when you get a min, can you kindly review and merge please?

File: website/docs/r/elasticache_serverless.html.markdown [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      unit    = "GB"
+    }
+    ecpu_per_second {
+      maximum = 5
+    }
+  }
+  daily_snapshot_time      = "09:00"
+  description              = "Test Server"
+  kms_key_id               = aws_kms_key.test.arn
+  major_engine_version     = "1.6"
```
Comment (noizo) [inline website/docs/r/elasticache_serverless.html.markdown:56]: should be something like 7.1 imo

**Comment by johnsonaj** [comment]
Hi @meetreks 👋🏾 , thank you for the contribution! A [policy change](https://github.com/hashicorp/terraform-provider-aws/issues/32917) was made earlier this year that now requires all net-new resources to be written using the [Terraform Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework). Since this is a new resource, it will need to be converted to using the Framework. You can always reference the [Contributor Guide](https://hashicorp.github.io/terraform-provider-aws/terraform-plugin-development-packages/#which-plugin-version-should-i-use) when choosing which plugin version to use. 

The [migration guild](https://hashicorp.github.io/terraform-provider-aws/migrate-from-sdk-to-framework/) should help with getting started when moving this SDKv2 resource to Framework. Please reach out if there are any issues.

File: website/docs/r/elasticache_serverless.html.markdown (hunk 1 above)
Comment (meetreks) [reply website/docs/r/elasticache_serverless.html.markdown]: Yeah -- you are right, its a copy pasta error butt fixed it now. I ran the testts with verssion 7 and amended to the same. For some reason when I ran the tests it said engine version 7.1 is not supported.

**Comment by meetreks** [comment]
@johnsonaj you mean I need to discard the current changes and regen using SDK V2?

File: website/docs/r/elasticache_serverless.html.markdown (hunk 1 above)
Comment (hazmei) [reply website/docs/r/elasticache_serverless.html.markdown]: @meetreks i believe it should be engine version 7. Was trying out awscc provider elasticache serverless resource earlier and it only accepted 7 not 7.1 for major engine version.

**Comment by noizo** [comment]
> @johnsonaj you mean I need to discard the current changes and regen using SDK V2?

I think it's about migrating from SDK V2 to https://developer.hashicorp.com/terraform/plugin/framework
Do You plan to continue work on it?

**Comment by meetreks** [comment]
> > @johnsonaj you mean I need to discard the current changes and regen using SDK V2?
> 
> I think it's about migrating from SDK V2 to https://developer.hashicorp.com/terraform/plugin/framework Do You plan to continue work on it?

Yes I am, but having issues with using tfsdk2fw to scaffold the v1 SDK to use new version of the plugin framework

**Comment by Menahem1** [comment]
Hello,
is there anything new ?

**Review by johnsonaj — APPROVED** [review]
LGTM 🚀 

```console
% make testacc TESTARGS='-run=TestAccElastiCacheServerlessCache_' PKG=elasticache

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/elasticache/... -v -count 1 -parallel 20  -run=TestAccElastiCacheServerlessCache_ -timeout 360m
--- PASS: TestAccElastiCacheServerlessCache_disappears (329.98s)
--- PASS: TestAccElastiCacheServerlessCache_fullRedis (341.09s)
--- PASS: TestAccElastiCacheServerlessCache_basic (343.06s)
--- PASS: TestAccElastiCacheServerlessCache_basicRedis (396.92s)
--- PASS: TestAccElastiCacheServerlessCache_full (420.86s)
--- PASS: TestAccElastiCacheServerlessCache_update (433.63s)
--- PASS: TestAccElastiCacheServerlessCache_tags (480.97s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/elasticache	484.543s
```

**Comment by aokellermann** [comment]
@johnsonaj thanks for your work on this! any idea when this feature will be released?

**Comment by johnsonaj** [comment]
@meetreks thank you for the contribution! 🎉

**Comment by minhhieu76qng** [comment]
@johnsonaj Thanks for you works. Could you tell us when this resource will be released?

**Comment by johnsonaj** [comment]
> @johnsonaj Thanks for your works. Could you tell us when this resource will be released?

Hi @minhhieu76qng. This resource will be available in this week's release, `v5.32.0`

**Comment by autarchprinceps** [comment]
A feature like name_prefix instead of hardcoding a name and having to delete then recreate would be helpful as well. 
Is there a particular reason the (read_)endpoint is a list, despite only ever containing one entry?

**Pull request merged**